packages/apollo-gateway/src/index.ts

```diff
@@ -19,50 +22,67 @@ import {
 } from './executeQueryPlan';
 
 import { getServiceDefinitionsFromRemoteEndpoint } from './loadServicesFromRemoteEndpoint';
+import { getServiceDefinitionsFromStorage } from './loadServicesFromStorage';
 
 import { serializeQueryPlan, QueryPlan } from './QueryPlan';
 import { GraphQLDataSource } from './datasources/types';
 import { RemoteGraphQLDataSource } from './datasources/RemoteGraphQLDatasource';
 import { HeadersInit } from 'node-fetch';
 
-export interface GraphQLService {
-  schema?: GraphQLSchema;
-  executor: GraphQLExecutor;
-  isReady: boolean;
-}
-
 export type ServiceEndpointDefinition = Pick<ServiceDefinition, 'name' | 'url'>;
 
-export interface GatewayConfigBase {
+interface GatewayConfigBase {
   debug?: boolean;
   // TODO: expose the query plan in a more flexible JSON format in the future
   // and remove this config option in favor of `exposeQueryPlan`. Playground
   // should cutover to use the new option when it's built.
   __exposeQueryPlanExperimental?: boolean;
   buildService?: (definition: ServiceEndpointDefinition) => GraphQLDataSource;
-  serviceList?: ServiceEndpointDefinition[];
+}
+
+interface RemoteGatewayConfig extends GatewayConfigBase {
+  serviceList: ServiceEndpointDefinition[];
   introspectionHeaders?: HeadersInit;
```

Comment:
I think we want to remove this bit and instead allow the `buildService` to be used to fetch the service definition

Reply:
Which part exactly?

Reply:
You mean configure introspection headers as part of `willSendRequest`, or in a new method? Or something else entirely?